e2e/cypress/integration/channel_sidebar/category_sorting_spec.js

```diff
@@ -0,0 +1,301 @@
+            testChannel = channel;
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T3834 Category sorting', () => {
+        const channelNames = [];
+        const categoryName = createCategoryFromSidebarMenu();
+
+        // # Create 5 channels and add them to a custom category
+        for (let i = 0; i < 5; i++) {
+            channelNames.push(createChannelAndAddToCategory(categoryName));
+            cy.get('#SidebarContainer .scrollbar--view').scrollTo('bottom', {ensureScrollable: false});
+        }
+
+        // # Sort alphabetically
+        cy.get(`.SidebarChannelGroupHeader:contains(${categoryName})`).within(() => {
+            // # Open dropdown next to channel name
+            cy.get('.SidebarMenu').invoke('show').get('.SidebarMenu_menuButton').should('be.visible').click({force: true});
+
+            // # Open sub menu
+            cy.get('#sortChannels').parent('.SubMenuItem').trigger('mouseover');
+
+            // # Click on sort alphabetically
+            cy.get('#sortAlphabetical').parent('.SubMenuItem').click();
+        });
+
+        // * Verify channels are sorted alphabetically
+        verifyAlphabeticalSortingOrder(categoryName, channelNames.length);
+
+        // # Add another channel
+        channelNames.push(createChannelAndAddToCategory(categoryName));
+        cy.get('#SidebarContainer .scrollbar--view').scrollTo('bottom', {ensureScrollable: false});
+
+        // * Verify channels are still sorted alphabetically
+        verifyAlphabeticalSortingOrder(categoryName, channelNames.length);
+
+        // # Sort by recency
+        cy.get(`.SidebarChannelGroupHeader:contains(${categoryName})`).within(() => {
+            // # Open dropdown next to channel name
+            cy.get('.SidebarMenu').invoke('show').get('.SidebarMenu_menuButton').should('be.visible').click({force: true});
+
+            // # Open sub menu
+            cy.get('#sortChannels').parent('.SubMenuItem').trigger('mouseover');
+
+            // # Click on sort by recency
+            cy.get('#sortByMostRecent').parent('.SubMenuItem').click();
+        });
+
+        // # Sort channel names in reverse order that they were created (ie. most recent to least)
+        let sortedByRecencyChannelNames = channelNames.concat().reverse();
+        for (let i = 0; i < channelNames.length; i++) {
+            // * Verify that the channels are in reverse order that they were created
+            cy.get(`.SidebarChannelGroup:contains(${categoryName}) .NavGroupContent li:nth-child(${i + 1}) a[id^="sidebarItem_${sortedByRecencyChannelNames[i]}"]`).should('be.visible');
+        }
+
+        // # Add another channel
+        channelNames.push(createChannelAndAddToCategory(categoryName));
+        cy.get('#SidebarContainer .scrollbar--view').scrollTo('bottom', {ensureScrollable: false});
+
+        // # Sort channel names in reverse order that they were created (ie. most recent to least)
```

Comment:
Normally yes, but you've pointed out that I'm definitely doing it wrong lol. Will have to fix that.

Reply:
Oh wait nevermind I'm fine lol. So that statement is just taking the list of channels in the order they were created, and copying that list into another list, but reversed. So we need to update that reversed list when we add a new channel to make sure it's up to date.

Reply:
Ah false alarm, yes disregard 
I was editing a step in the test case to match this and it felt redundant